sourcegraph/sourcegraph
# web: add app mount shared span

## Context

This PR adds another shared span — `AppMount`- representing the React application mount event. It will be used to calculate time from React app mount until the target React component mount. E.g., how much time on average do we wait for the `Blob` component to mount after `SourcegraphWeb` did mount?

Partially addresses https://github.com/sourcegraph/sourcegraph/issues/44008.

## Test plan

1. `ENABLE_OPEN_TELEMETRY=true sg start web-standalone`
2. Open the browser console and ensure that debug-level output is enabled.
3. Reload any web application page.
4. See the app mount span attached to the window-load span.

<img width="298" alt="Screenshot 2022-11-07 at 16 42 30" src="https://user-images.githubusercontent.com/3846380/200264820-84006277-f8ab-4c04-9d8d-c7b9e39415ad.png">

## App preview:

- [Web](https://sg-web-vb-add-app-mount-span.onrender.com/search)

Check out the [client app preview documentation](https://docs.sourcegraph.com/dev/how-to/client_pr_previews) to learn more.
